src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectDependencyProvider.cs

```diff
@@ -0,0 +1,155 @@
+                    {
+                        packageInfo[packageItem.EvaluatedInclude] = packageDependency;
+                    }
+                }
+
+                // 2nd pass to populate dependencies;
+
+                PopulateDependencies(packageInfo, packageItems);
+            }
+
+            return packageInfo.Values;
+        }
+
+
+        public IEnumerable<ResolvedReference> GetResolvedReferences()
+        {
+            var refItems = _projectInstance.GetItems(ResolvedReferenceItemType);
+            var resolvedReferences = new List<ResolvedReference>();
+            if (refItems != null)
+            {
+                foreach (var refItem in refItems)
+                {
+                    resolvedReferences.Add(CreateResolvedReferenceFromProjectItem(refItem));
+                }
+            }
+
+            return resolvedReferences;
+        }
+
+        private static ResolvedReference CreateResolvedReferenceFromProjectItem(ProjectItemInstance item)
+        {
+            string itemSpec = string.Empty;
+            string nugetPackageSource = string.Empty;
+            string version = string.Empty;
+            string resolvedPath = string.Empty;
+            string name = string.Empty;
+
+            resolvedPath = item.EvaluatedInclude;
+
+            name = resolvedPath ?? Path.GetFileNameWithoutExtension(resolvedPath);
+
+            foreach (var m in item.Metadata)
+            {
+                if (ItemSpecKey.Equals(m.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    itemSpec = m.EvaluatedValue;
+                    continue;
+                }
+
+                if (NugetSourceTypeKey.Equals(m.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    nugetPackageSource = m.EvaluatedValue;
+                    continue;
+                }
+
+                if (VersionKey.Equals(m.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    version = m.EvaluatedValue;
+                }
+            }
+
+            var isFromNugetPackage = "Package".Equals(nugetPackageSource, StringComparison.OrdinalIgnoreCase);
+            return new ResolvedReference(name, version, resolvedPath, isFromNugetPackage);
+        }
+
+        private static DependencyDescription CreateDependencyDescriptionFromItem(ProjectItemInstance item)
+        {
+            var version = item.GetMetadataValue("Version");
+            var path = item.GetMetadataValue("Path");
+            var type = item.GetMetadataValue("Type");
+            var resolved = item.GetMetadataValue("Resolved");
+            var itemSpec = item.EvaluatedInclude;
+
+            // For type == Target, we do not get Name in the metadata. This is a special node where the dependencies are 
+            // the direct dependencies of the project.
+
+            var name = item.HasMetadata("Name") ? item.GetMetadataValue("Name") : itemSpec;
```

Comment:
AFAIK, `item.GetMetadataValue("Name")`  would return an empty string if the there is no metadata with key as `"Name"`